x/bep3/keeper/swap.go

```diff
@@ -0,0 +1,208 @@
+package keeper
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/bep3/types"
+)
+
+// CreateAtomicSwap creates a new AtomicSwap
+func (k Keeper) CreateAtomicSwap(ctx sdk.Context, randomNumberHash []byte, timestamp int64, heightSpan int64,
+	sender sdk.AccAddress, recipient sdk.AccAddress, senderOtherChain, recipientOtherChain string,
+	amount sdk.Coins, expectedIncome string) sdk.Error {
+
+	swapID := types.CalculateSwapID(randomNumberHash, sender, senderOtherChain)
+
+	// Confirm that this swap is valid
+	_, found := k.GetAtomicSwap(ctx, swapID)
+	if found {
+		return types.ErrAtomicSwapAlreadyExists(k.codespace, swapID)
+	}
+
+	// The heightSpan period should be more than 10 minutes and less than one week
+	// Assume average block time interval is 10 second. 10 mins = 60 blocks, 1 week = 60480 blocks
+	if heightSpan < k.GetMinBlockLock(ctx) || heightSpan > k.GetMaxBlockLock(ctx) {
+		return types.ErrInvalidHeightSpan(k.codespace, heightSpan, k.GetMinBlockLock(ctx), k.GetMaxBlockLock(ctx))
+	}
+
+	// Validate that timestamp is within reasonable bounds
+	// Assuming a block time of 10 seconds, the timestamp must be in range [-15 mins, 30 mins] of the current time
```

Comment:
Confused as to what this validation is checking. It says it's checking the timestamp (which I assume is a unix timestamp like 1582914311), but is then checking the block height. Clarify in comment perhaps.

Reply:
Updated comment to clarify that timestamp represents block height in https://github.com/Kava-Labs/kava/pull/370/commits/ac92e26999cf32a5fae8296ba92a05b1da0cf357. I agree that this is confusing and will make a note to check with Binance to confirm that this implementation of timestamp is correct.

Reply:
This also assumes a block time of 10 seconds as opposed to 7 seconds and should be updated to match the new 7-second standard.